cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -916,6 +899,16 @@ func (m *McmManager) GetMachineDeploymentNodeTemplate(machinedeployment *Machine
 	return nodeTmpl, nil
 }
 
+// GetMachineDeploymentResource returns the MachineDeployment object for the provided machine deployment name
+func (m *McmManager) GetMachineDeploymentResource(mdName string) (*v1alpha1.MachineDeployment, error) {
```

Comment:
```suggestion
func (m *McmManager) GetMachineDeploymentObject(mdName string) (*v1alpha1.MachineDeployment, error) {
```
Ig `Object` is more accurate then `Resource`?